drivers/spi/spi_nrfx_spi.c

```diff
@@ -383,7 +381,10 @@ static int spi_nrfx_pm_action(const struct device *dev,
 	};								       \
 	IF_ENABLED(CONFIG_PINCTRL, (PINCTRL_DT_DEFINE(SPI(idx))));	       \
 	static const struct spi_nrfx_config spi_##idx##z_config = {	       \
-		.spi = NRFX_SPI_INSTANCE(idx),				       \
+		.spi = {						       \
+			.p_reg = (NRF_SPI_Type *)DT_REG_ADDR(SPI(idx)),	       \
+			.drv_inst_idx = NRFX_SPI##idx##_INST_IDX,	       \
```

Comment:
I'm not convinced about these changes. They touch kind of internal structures of nrfx drivers that are supposed to be filled with the `NRFX_*_INSTANCE` macros. And for several single-instance nrfx drivers (like I2S, PDM, or USBD) we won't be able to use addresses from devicetree anyway, because they are currently hardcoded in those drivers.
I'm not sure if it is worth it.

Reply:
I agree it's not ideal, I'd expect a macro that allows to pass reg.

Reply:
The improvement is to align with the way Zephyr works, ie, hardware is described in Devicetree. With current design, we're leveraging this to the HAL, that is, HAL is driving driver's design. This approach is problematic, the best example being the instantiation.